### PR TITLE
Implement grip pose input action for Oculus Touch controller's

### PR DESCRIPTION
**Pull request by FejZa**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Implemented grip pose input action for Oculus Touch controllers. Useful when grabbing something and looking to "attach" it to the controller's / hand's grip pose.

## Submodule Changes

- [Core](https://github.com/XRTK/XRTK-Core/pull/718)
- [SDK](https://github.com/XRTK/SDK/pull/211)

### CODE REVIEW

**Comment by SimonDarksideJ** [comment]
The pr only seems to apply the current controller pose to the grip pose, is this correct? As I would have expected some grip offset?

**Comment by FejZa** [comment]
Correct. In my tests the current controller pose seemed pretty much right.

**Comment by FejZa** [comment]
If you want I can add an option for a configurable? offset

**Comment by SimonDarksideJ** [comment]
I think that would be best, else there isn't much point just duplicating the current pose

**Comment by FejZa** [comment]
I can add it. It's still adding value though as while in code it's currently the same pose as the controller's, semantically it's another thing from a user's perspective. A user doesn't care where the pose comes from. He cares that it's the grip pose.

I'll add an offset option though, i agree it makes sense.

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/Controllers/BaseOculusController.cs [hunk 1]
```diff
@@ -28,6 +28,13 @@ protected BaseOculusController(IMixedRealityControllerDataProvider controllerDat
             NodeType = nodeType;
         }
 
+        /// <summary>
+        /// Local offset from the controller position defining where the grip pose is.
+        /// The grip pose may be used to attach things to the controller when grabbing objects.
+        /// </summary>
+        private readonly MixedRealityPose gripPoseOffset = new
```
Comment (SimonDarksideJ) [inline XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/Controllers/BaseOculusController.cs:35]: Can we move this variable to the base controller definition? Would seem to make sense to make this a generic thing as it is likely to be needed on multiple platforms.

**Review by SimonDarksideJ — CHANGES_REQUESTED** [review]

File: XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/Controllers/BaseOculusController.cs (hunk 1 above)
Comment (FejZa) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/Controllers/BaseOculusController.cs:35]: I thought about that. But essentially I don't think every controller has a grip pose, right?

Comment (FejZa) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/Controllers/BaseOculusController.cs:35]: E.g. for the Oculus Remote it does not make much sense?

Comment (StephenHodgson) [reply XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/Controllers/BaseOculusController.cs:35]: I think we only had a pointer pose offset, but I suppose a grip pose offset would be useful in the base controller definition

**Review by StephenHodgson — DISMISSED** [review]

**Comment by FejZa** [comment]
I think I might need to update the Core package dependency for this to succeed. Because it needs the property in `BaseController`

**Review by SimonDarksideJ — APPROVED** [review]

**Review by SimonDarksideJ — APPROVED** [review]

**Pull request merged**